backend/src/main/java/com/htecgroup/skynest/service/impl/RefreshTokenServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.htecgroup.skynest.service.RefreshTokenService;
+import com.htecgroup.skynest.service.UserService;
+import com.htecgroup.skynest.util.JwtUtils;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.modelmapper.ModelMapper;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.htecgroup.skynest.util.JwtUtils.ACCESS_TOKEN_EXPIRATION_MS;
+import static com.htecgroup.skynest.util.JwtUtils.ALGORITHM;
+import static java.util.Arrays.stream;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class RefreshTokenServiceImpl implements RefreshTokenService {
+
+  private UserService userService;
+  private ModelMapper modelMapper;
+
+  @Override
+  public void refreshToken(HttpServletRequest request, HttpServletResponse response) {
+    String header = request.getHeader("refresh-token");
```

Comment:
You method should not be dependent on the request and response. The logic of the method is to refresh the token, meaning that you need the logged user as input param, and custom object that will contain the new attributes (new auth token, and refresh token ) . Then in the constructor you will apply that result in the header or body or how ever you need it.
Btw logged user you can see how to get it with the new change that @MarkoRadunovicHtec  did

Reply:
I finally did it, it should be fine now.